src/Gui/MainWindow.cpp

```diff
@@ -2079,6 +2081,51 @@ void MainWindow::setPaneText(int i, QString text)
     }
 }
 
+void MainWindow::initialiseSizeLabel()
+{
+    d->sizeLabel = new QToolButton(statusBar());
+    d->sizeLabel->setText(tr("Dimension"));
+    d->sizeLabel->setPopupMode(QToolButton::MenuButtonPopup);
+    QObject::connect(
+        d->sizeLabel, &QToolButton::clicked,
+        d->sizeLabel, &QToolButton::showMenu
+    );
+    d->sizeLabel->setMinimumWidth(120);
+
+    //create the button actions
+    int num = static_cast<int>(Base::UnitSystem::NumUnitSystemTypes);
+
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Units");
+    int userSchema = hGrp->GetInt("UserSchema", 0);
+
+    for (int i = 0; i < num; i++) {
+        QAction* action = new QAction(qApp->translate("Gui::Dialog::DlgSettingsUnits", Base::UnitsApi::getDescription(static_cast<Base::UnitSystem>(i))), this);
+        action->setCheckable(true);
+        action->setChecked(i == userSchema);
+        d->sizeLabel->addAction(action);
+
+        QObject::connect(qAsConst(d->sizeLabel)->actions()[i], &QAction::changed,
```

Comment:
Isn't `d->sizeLabel)->actions()[i]` the same as `action`?

Reply:
Yes this is correct thanks.